providers/smtp/src/airflow/providers/smtp/notifications/smtp.py

```diff
@@ -106,39 +110,44 @@ def hook(self) -> SmtpHook:
         """Smtp Events Hook."""
         return SmtpHook(smtp_conn_id=self.smtp_conn_id, auth_type=self.auth_type)
 
-    def notify(self, context):
+    def _build_email_content(self, smtp: SmtpHook, context: Context, use_templates: bool = True):
+        # TODO:  use_templates is temporary until templating on the Triggerer is sorted out.
+        #   Once that is done, we can remove that flag.
+        fields_to_re_render = []
+        if self.from_email is None:
+            if smtp.from_email is not None:
+                self.from_email = smtp.from_email
+            else:
+                raise ValueError("You should provide `from_email` or define it in the connection")
+            fields_to_re_render.append("from_email")
+        if self.subject is None:
+            smtp_default_templated_subject_path: str
+            if smtp.subject_template:
+                smtp_default_templated_subject_path = smtp.subject_template
+            else:
+                smtp_default_templated_subject_path = (
+                    Path(__file__).parent / "templates" / "email_subject.jinja2"
+                ).as_posix()
+            self.subject = self._read_template(smtp_default_templated_subject_path)
+            fields_to_re_render.append("subject")
+        if self.html_content is None:
+            smtp_default_templated_html_content_path: str
+            if smtp.html_content_template:
+                smtp_default_templated_html_content_path = smtp.html_content_template
+            else:
+                smtp_default_templated_html_content_path = (
+                    Path(__file__).parent / "templates" / "email.html"
+                ).as_posix()
+            self.html_content = self._read_template(smtp_default_templated_html_content_path)
+            fields_to_re_render.append("html_content")
+        if fields_to_re_render and use_templates:
+            jinja_env = self.get_template_env(dag=context["dag"])
```

Comment:
I think this should fix the issue causing templating to crash with the simplified Dag run-only context:
```suggestion
            jinja_env = self.get_template_env(dag=context.get("dag"))
```